File: app/src/main/java/dev/olog/basil/presentation/ingredients/IngredientsAdapter.java
package dev.olog.basil.presentation.ingredients;

import java.util.Collections;
import java.util.List;

import androidx.databinding.ViewDataBinding;
import dev.olog.basil.BR;
import dev.olog.basil.R;
import dev.olog.basil.presentation.base.BaseAdapter;
import dev.olog.basil.presentation.model.DisplayableIngredient;

public class IngredientsAdapter extends BaseAdapter<DisplayableIngredient> {

    @Override
    protected void initViewHolderListeners(ViewDataBinding binding) {

    }

    @Override
    protected void bind(ViewDataBinding binding, DisplayableIngredient item, int position) {
        binding.setVariable(BR.item, item);
    }

    @Override
    protected int provideLayoutId() {
        return R.layout.item_ingredient;
    }

    @Override
    public void updateDataSet(List<DisplayableIngredient> newData) {
        Collections.sort(newData, (o1, o2) -> o1.getOrder() - o2.getOrder());
        super.updateDataSet(newData);
    }
}

File: app/src/main/java/dev/olog/basil/presentation/detail/adapter/IngredientsAdapter.kt
package dev.olog.basil.presentation.detail.adapter

import androidx.databinding.ViewDataBinding
import dev.olog.basil.BR
import dev.olog.basil.R
import dev.olog.basil.model.Ingredient
import dev.olog.basil.presentation.base.BaseAdapter

class IngredientsAdapter : BaseAdapter<Ingredient>() {

    override fun initViewHolderListeners(binding: ViewDataBinding) {

    }

    override fun bind(binding: ViewDataBinding, item: Ingredient, position: Int) {
        binding.setVariable(BR.item, item)
    }

    override fun provideLayoutId(): Int {
        return R.layout.item_ingredient
    }


    override fun updateDataSet(newData: List<Ingredient>) {
        val data = newData.sortedBy { it.order }
        super.updateDataSet(data)
    }
}
